### PR TITLE
[Heartbeat] Add Magefile to X-Pack

### PR DESCRIPTION
**Pull request by andrewvc**
This also restructures the main magefile to share code with the x-pack. 

This fixes the broken agent build which is trying to run `mage build` in the x-pack heartbeat folder.

### CODE REVIEW

File: heartbeat/magefile.go [hunk 1]
```diff
@@ -23,12 +23,10 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/magefile/mage/mg"
-	"github.com/magefile/mage/sh"
-
 	devtools "github.com/elastic/beats/v7/dev-tools/mage"
 	"github.com/elastic/beats/v7/generator/common/beatgen"
 	heartbeat "github.com/elastic/beats/v7/heartbeat/scripts/mage"
+	"github.com/magefile/mage/mg"
```
Comment (kvch) [inline heartbeat/magefile.go:29]: This import seems to be at incorrect place.

File: heartbeat/magefile.go [hunk 2]
```diff
@@ -107,14 +105,13 @@ func TestPackages() error {
 	return devtools.TestPackages(devtools.WithMonitorsD())
 }
 
-// Update updates the generated files (aka make update).
-func Update() error {
-	return sh.Run("make", "update")
+func Fields () error {
```
Comment (kvch) [inline heartbeat/magefile.go:108]: There is a space between the function name and `()`.

**Comment by andrewvc** [comment]
Thanks @kvch pushed some fixes and re-ran `mage fmt` in both spots.

File: x-pack/heartbeat/main.go [hunk 3]
```diff
@@ -1,6 +1,19 @@
-// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```
Comment (kvch) [inline x-pack/heartbeat/main.go:1]: This licence header change seems incorrect.

File: x-pack/heartbeat/main_test.go [hunk 4]
```diff
@@ -1,6 +1,20 @@
-// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```
Comment (kvch) [inline x-pack/heartbeat/main_test.go:1]: This licence header change seems incorrect.

File: x-pack/heartbeat/magefile.go [hunk 5]
```diff
@@ -0,0 +1,131 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
```
Comment (kvch) [inline x-pack/heartbeat/magefile.go:1]: This licence is incorrect as well.

File: x-pack/heartbeat/cmd/root.go [hunk 6]
```diff
@@ -1,6 +1,19 @@
-// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
-// or more contributor license agreements. Licensed under the Elastic License;
-// you may not use this file except in compliance with the Elastic License.
+// Licensed to Elasticsearch B.V. under one or more contributor
```
Comment (kvch) [inline x-pack/heartbeat/cmd/root.go:1]: Incorrect licence.

**Pull request merged**